cache/cache.go

```diff
@@ -331,14 +331,19 @@ func (dc *directoryCache) Add(key string, opts ...Option) (Writer, error) {
 		abortFunc: func() error {
 			defer w.Close()
 			defer w.Abort()
-			dc.bufPool.Put(b) // abort it.
+			dc.putBuffer(b) // abort it.
 			return nil
 		},
 	}
 
 	return memW, nil
 }
 
+func (dc *directoryCache) putBuffer(b *bytes.Buffer) {
+	b.Reset()
```

Comment:
Note that calling Reset() is the safest but may not be the most efficient solution.

Go's standard library has some more sophisticated examples.

https://github.com/golang/go/blob/9e7bc80b31088dc62faf4776ffdb1a2e27afa94e/src/os/dir_unix.go#L30-L36

https://github.com/golang/go/blob/acc2957bc9873ab7e65942045830a3dc0592eda2/src/net/http/server.go#L794-L850